File: docgen/src/lib/codeParser.js
// Function to parse code from provided URLs
async function parseCode(urls) {
    const codeData = [];
    console.log(`Parsing code from provided URLs...`);
  
    for (const url of urls) {
      // Check if the file is one of the supported formats
      if (url.match(/\.(js|jsx|ts|tsx|py|java|html|css)$/)) {
        console.log(`Fetching content from: ${url}`);
        try {
          const response = await fetch(url);
          const content = await response.text();
          codeData.push({
            file_path: url,
            content: content
          });
        } catch (error) {
          console.error(`Error fetching the file from ${url}:`, error);
        }
      } else {
        console.warn(`Unsupported file type for URL: ${url}`);
      }
    }
  
    console.log(`Total files parsed: ${codeData.length}`);
    return codeData;
  }
  
  // Example usage:
  // const urls = ["https://example.com/script.js", "https://example.com/styles.css"];
  // parseCode(urls).then(data => console.log(data));
  

File: code_parser.py
import os

def parse_code(path):
    code_data = []
    print(f"Searching for files in {path}")
    for root, dirs, files in os.walk(path):
        for file in files:
            if file.endswith(('.js', '.jsx', '.ts', '.tsx', '.py', '.java', '.html', '.css')):
                file_path = os.path.join(root, file)
                print(f"Found file: {file_path}")
                with open(file_path, 'r', encoding='utf-8', errors='ignore') as f:
                    content = f.read()
                code_data.append({
                    'file_path': file_path,
                    'content': content
                })
    print(f"Total files found: {len(code_data)}")
    return code_data
